cmd/boulder-wfe/main.go

```diff
@@ -18,8 +20,55 @@ import (
 
 const clientName = "WFE"
 
-func setupWFE(c cmd.Config, logger blog.Logger, stats metrics.Statter) (*rpc.RegistrationAuthorityClient, *rpc.StorageAuthorityClient) {
-	amqpConf := c.WFE.AMQP
+type config struct {
+	WFE struct {
+		cmd.ServiceConfig
+		BaseURL       string
+		ListenAddress string
+
+		AllowOrigins []string
+
+		CertCacheDuration           string
+		CertNoCacheExpirationWindow string
+		IndexCacheDuration          string
+		IssuerCacheDuration         string
+
+		ShutdownStopTimeout string
+		ShutdownKillTimeout string
+
+		SubscriberAgreementURL string
+	}
+
+	*cmd.AllowedSigningAlgos
+
+	cmd.StatsdConfig
+
+	SubscriberAgreementURL string
+
+	cmd.SyslogConfig
+
+	Common struct {
+		BaseURL    string
+		IssuerCert string
+	}
+}
+
+func (cfg config) KeyPolicy() goodkey.KeyPolicy {
+	if cfg.AllowedSigningAlgos != nil {
+		return goodkey.KeyPolicy{
+			AllowRSA:           cfg.AllowedSigningAlgos.RSA,
+			AllowECDSANISTP256: cfg.AllowedSigningAlgos.ECDSANISTP256,
+			AllowECDSANISTP384: cfg.AllowedSigningAlgos.ECDSANISTP384,
+			AllowECDSANISTP521: cfg.AllowedSigningAlgos.ECDSANISTP521,
+		}
+	}
+	return goodkey.KeyPolicy{
+		AllowRSA: true,
+	}
+}
```

Comment:
The way we've dealth with similar cases is to define a method on the `AllowedSigningAlgos` type that generates the desired output, in this case KeyPolicy. I think you can even make it return correct results when called on a nil receiver if you're careful.